setuptools/msvc.py

```diff
@@ -212,6 +219,19 @@ def msvc14_get_vc_env(plat_spec):
         raise
 
 
+def msvc14_gen_lib_options(*args, **kwargs):
+    """
+    Patched "distutils._msvccompiler.gen_lib_options" for fix 
+    compatibility between "numpy.distutils" and "distutils._msvccompiler"
+    (for Numpy < 1.11.2)
+    """
+    if "numpy" in distutils.ccompiler.CCompiler.spawn.__module__:
```

Comment:
That's very specific. Testing for `"numpy.distutils" in sys.modules` sounds less hackish, no?